runtime/parser/declaration.go

```diff
@@ -1157,10 +1212,16 @@ func parseSpecialFunctionDeclaration(
 	functionBlockIsOptional bool,
 	access ast.Access,
 	accessPos *ast.Position,
+	purity ast.FunctionPurity,
+	purityPos *ast.Position,
 	identifier ast.Identifier,
 ) (*ast.SpecialFunctionDeclaration, error) {
 
 	startPos := identifier.Pos
+	// access modifier will precede purity if both exist
+	if purityPos != nil {
+		startPos = *purityPos
+	}
 	if accessPos != nil {
```

Comment:
If both purity and access modifier are specified, the latter will override the former:

```suggestion
	} else if accessPos != nil {
```

Reply:
This is intentional: if both purity and access are specified on a function declaration, we want to use the earlier of the two to compute the start of the function declaration. That said I have a more organized way of doing this that I will update to.